### PR TITLE
GEOPY-1501: Speed up Peak merging

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-1501 - Speed up Peak merging**

### CODE REVIEW

File: peak_finder/line_group.py [hunk 1]
```diff
@@ -230,9 +230,50 @@ def get_anomaly_attributes(self, line_data_subset: list[LineData]) -> tuple:
...
+        for next_neighbour in next_neighbours:
+            current = np.vstack([path, neighbourhood[next_neighbour]])
+            paths = self.accumulate_groups(current, neighbourhood)
+
+            for branch in paths:
+                branches.append(branch)
+
+        return branches
+
+    def find_neighbour_groups(self, groups: list[AnomalyGroup]) -> list:
```
Comment (benk-mira) [inline peak_finder/line_group.py:255]: unit test

File: peak_finder/application.py [hunk 2]
```diff
@@ -717,25 +718,23 @@ def compute_lines(  # pylint: disable=too-many-arguments, too-many-locals
...
+            if "n_lines" in triggers and line_anomaly.line_id in self.computed_lines:
+                continue
+
             line_groups = line_anomaly.anomalies
             line_anomalies: list[AnomalyGroup] = []
             if line_groups is not None:
                 for line_group in line_groups:
                     line_anomalies += line_group.groups
+
             if line_anomaly.line_id not in self.computed_lines:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply peak_finder/application.py:731]: The way it is written right now, we are appending anomalies from different property_groups to the same line id dict. So this loop can be called multiple times, but we don't want to reset it everytime

File: peak_finder/line_group.py (hunk 1 above)
Comment (domfournier) [reply peak_finder/line_group.py]: I haven't done a full unit test for it as it is pretty involved. Need to create line data, then anomalies. It sort of baked in the peak finder tests though...

Comment (domfournier) [reply peak_finder/line_group.py]: Did do a big cleanup of the AnomalyGroup class though

File: peak_finder/application.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (benk-mira) [reply peak_finder/application.py:731]: ok

File: peak_finder/line_group.py (hunk 1 above)
Comment (benk-mira) [reply peak_finder/line_group.py]: sure

**Review by benk-mira — APPROVED** [review]

**Pull request merged**